### PR TITLE
multi-path creation in explorer: validation detects if file is not leaf

### PR DESCRIPTION
**Pull request by tsalinger**
Previously, the validation would not detect if an user tries to create or rename & move an item as a descendant of a file

### CODE REVIEW

**Comment by isidorn** [comment]
@tsalinger thanks for your PR. However this milestone we finish on wednesday and I unfortunetly will not have time to review until then - sorry about that but I also do not think this is very critical. 
@bpasero are you maybe interested in reviewing or we push this to next milestone?

**Comment by bpasero** [comment]
@isidorn I trust your capable review skills

**Comment by tsalinger** [comment]
@isidorn no problem, I hope you'll get your things done in time.
I refactored the PR a bit. Take your time reviewing it. 
In the meantime, I'll push the open things to the next milestone.

**Comment by isidorn** [comment]
@tsalinger I took a look at the PR and it looks good to me.
Did you test it out with some corner cases?

Also I left some comments in the code

File: src/vs/workbench/parts/files/electron-browser/fileActions.ts [hunk 1]
```diff
@@ -1392,6 +1385,33 @@ export function validateFileName(parent: ExplorerItem, name: string, allowOverwr
...
+
+interface IMappedPath {
+	fullPathAlreadyExists: boolean;
+	lastExistingPathSegment: {
+		isFile: boolean;
+		name: string;
+	};
+}
+
+function mapPathsToExistingFolders(parent: ExplorerItem, pathNames: string[]): IMappedPath {
```
Comment (isidorn) [inline src/vs/workbench/parts/files/electron-browser/fileActions.ts]: Not very happy about the names here. When I read "map" I expect something like an actual map.
I would rename to something simplier, maybe something with "check" and "path". I leave this up to you

File: src/vs/workbench/parts/files/electron-browser/fileActions.ts [hunk 2]
```diff
@@ -1392,6 +1385,33 @@ export function validateFileName(parent: ExplorerItem, name: string, allowOverwr
 	return null;
 }
 
+
+interface IMappedPath {
```
Comment (isidorn) [inline src/vs/workbench/parts/files/electron-browser/fileActions.ts]: Having an interface just for the return type which is used in one place feels like overkill. I would just have this inline.

**Comment by tsalinger** [comment]
@isidorn Thanks for the feedback.
I tested it by hand and couldn't make it fail.
In addition I added some unit tests. They might come in handy in the future.

**Comment by isidorn** [comment]
Thanks for adding tests, looks good to me. Merging in
